addon/mongocrypt.h

```diff
@@ -62,7 +62,15 @@ class MongoCrypt : public Napi::ObjectWrap<MongoCrypt> {
    public:
     static Napi::Function Init(Napi::Env env);
 
+    struct State {
+        std::unique_ptr<CryptoHooks> crypto_hooks;
+        std::unique_ptr<mongocrypt_t, MongoCryptDeleter> mongo_crypt;
+        MongoCrypt* js_wrapper;
```

Comment:
Just want to make sure I understand js_wrapper: this is necessary to obtain the `Env` object needed to interact with JS-methods (such as invoking the logger or converting Binarys to Buffers) but only for a handful of methods, where a reference to a method that needs the env potentially escapes the lifetime of the MongoCrypt object?  The logger, for example, can be called by libmongocrypt even after a MongoCrypt is destroyed.  And crypto hooks could theoretically be invoked by libmongocrypt after a MongoCrypt has been garbage collected.  

The js_wrapper construct provides a way to deteremine if the JS-exposed MongoCrypt object has been garbage collected when one of these methods is invoked.

Is that correct?

Reply:
> this is necessary to obtain the `Env` object needed to interact with JS-methods

Yeah, that's currently all it's used for – to reach back into JS.

> but only for a handful of methods

This should apply to any callbacks from C++ back into JS – are there any exceptions I'm missing?

> The logger, for example, can be called by libmongocrypt even after a MongoCrypt is destroyed. And crypto hooks could theoretically be invoked by libmongocrypt after a MongoCrypt has been garbage collected.

Yeah, because the underlying `mongocrypt_t` is still potentially alive and accessible via the `MongoCryptContext` instances.

> The js_wrapper construct provides a way to deteremine if the JS-exposed MongoCrypt object has been garbage collected when one of these methods is invoked.

I'd maybe say "a way to determine whether the JS object is still usable", it's not about garbage collection specifically (there are other cases, the most prominent one being the shutdown process of a Node.js Worker thread).